url78eec0e0b6_Orikuro_vic_tgz-pJ8-net_nexon_vindictus_itemcomparer_inport_StaticImportJ8/src/net/nexon/vindictus/itemcomparer/inport/StaticImport.java

```diff
@@ -0,0 +1,2125 @@
+		gloves.add(g);
+		pants.add(p);
+		armors.add(a);
+		helms.add(h);
+	}
+	
+	public static void staticImAndExport() {
+		// 60 sets
+		generateWonderland();
+		generateLightwind();
+		generateRaiders();
+		generateBlackHammer();
+		generateShiningWill();
+		generateSwift();
+		generateSilentQue();
+		generateHolyWing();
+
+		// 70 sets
+		generateChampion();
+		generateChampionSoft();
+		generateImperialHard();
+		generateImperialSoft();
+		generateRoyalGHeavy();
+		generateMhara();
+		generateEverlastingEvie();
+		generateApocryphal();
+		generateEverlastingFio();
+		generateNighthawk();
+		generatePhantomHeavy();
+		generateRoyalGuard();
+		generateNighthawkLight();
+		generatePhantomLight();
+
+		// 80 sets
+		generateSkeleton();
+		generateSkeletonWitch();
+		generatePearlClassic();
+		generateBlackPearl();
+		generateHeremon();
+		generateNoblesse();
+		generateHeremonSoft();
+		generatePirateLeather();
+		generatePirate();
+		generateNobleseWitch();
+
+		// neue sets
+		generateWEdge();
+		generateK9();
+		generateHMagnum();
+
+		// Episode 4 Season 2
+		generateMajest();
+		generateMajestLeather();
+		generateMajestSoft();
+		generateArmageddon();
+		generateArmageddonWitch();
+
+		// Lynn
+		generateSoulloop();
+		generateOriental();
+		generateDancing();
+
+		// arisha
+		generateApex();
+		generateSilverFox();
+		generateStriker();
+
+		// 90 gear
+		generateRegina_3_min();
+		generateBarha_3_min();
+		generateGuardian_3_min();
+		generateCloth_3_min();
+	}
+
+	public static void staticExportSets() {
+		HashSet<ItemSet> setnames = new HashSet<>();
+
+		for (Shoes s : shoes) {
+			setnames.add(s.getItemset());
+		}
+		for (Pants s : pants) {
+			setnames.add(s.getItemset());
+		}
+		for (Gloves s : gloves) {
+			setnames.add(s.getItemset());
+		}
+		for (Armor s : armors) {
+			setnames.add(s.getItemset());
+		}
+		for (Helm s : helms) {
+			setnames.add(s.getItemset());
+		}
+		isets = setnames;
+		try {
+			File outd = new File("data" + File.separator);
+			outd.mkdirs();
+			BufferedWriter out = new BufferedWriter(
+					new OutputStreamWriter(new FileOutputStream("data"
+							+ File.separator + "_sets.xml"), "UTF-8"));
+
+			JAXBContext context = JAXBContext.newInstance(ItemSets.class);
+			Marshaller m = context.createMarshaller();
+			m.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+			m.marshal(new ItemSets(setnames), out);
+			out.close();
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+
+	}
+
+	private static void exportPricesCSV() {
+		StringBuilder sb = new StringBuilder();
+
+		sb.append("id;price;Typ;Set;Itemname;Level;Stars;Useful;Armortype;");
+		sb.append("\n");
+
+		for (Shoes s : shoes) {
+			String x = s.getPrice() + "";
+			x = x.replace(".", ",");
+			sb.append(s.getId() + ";" + x + ";Shoes;"
+					+ s.getItemset().getName() + ";" + s.getName() + ";"
+					+ s.getLevel() + ";" + s.getStars() + ";"
+					+ s.getItemset().getUsefulChars() + ";" + s.getTyp());
+			sb.append("\n");
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 1976:1984, 1985:1993, 1994:2002, 2003:2011, 2012:2020.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.